src/libbson/tests/test-bson.c

```diff
@@ -1119,6 +1119,18 @@ test_bson_validate_dbpointer (void)
 }
 
 
+static void
+test_bson_validate_with_error_and_offset (void)
+{
+   size_t err_offset = 12345;
+   bson_error_t err = {67890};
+   bson_t bson = {0};
+   ASSERT (!bson_validate_with_error_and_offset (&bson, BSON_VALIDATE_NONE, &err_offset, &err));
+   ASSERT (err_offset == 0);
+   ASSERT (err.domain != 67890);
```

Comment:
```suggestion
   ASSERT_CMPSIZE_T (err_offset, ==, 0);
   ASSERT_CMPUINT32 (err.domain, !=, 67890); // domain is overwritten.
```

Suggest using `ASSERT_CMP*`. On failure, the compared numbers are printed:

```
Assert Failure: 0 == 123
```